plugins/modules/aci_esg.py

```diff
@@ -0,0 +1,375 @@
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import (
+    ACIModule,
+    aci_argument_spec,
+)
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        ap=dict(type="str", aliases=["app_profile", "app_profile_name"]),
+        esg=dict(type="str", aliases=["name", "esg_name"]),
+        admin_state=dict(type="str", choices=["no", "yes"]),  # ESG Admin State
+        vrf=dict(type="str", aliases=["vrf_name"]),  # ESG VRF name
+        description=dict(type="str", aliases=["descr"]),
+        intra_esg_isolation=dict(
+            type="str",
+            choices=["enforced", "unenforced"],
+        ),  # Intra ESG Isolation
+        preferred_group_member=dict(type="str", choices=["exclude", "include"]),  # Preferred Group Member
+        state=dict(
+            type="str",
+            default="present",
+            choices=["absent", "present", "query"],
+        ),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "ap", "esg"]],
+            ["state", "present", ["tenant", "ap", "esg"]],
```

Comment:
I would add vrf to the state present requirements as the UI does require the VRF to be set.